spec/views/ui-components/v1/cards/_eligibility_check.html.erb_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+RSpec.describe "ui-components/v1/cards/_eligibility_check.html.slim" do
+  context "when the eligibility partial is rendered" do
+
+    let(:employer_profile) do
+      instance_double(
+        "EmployerProfile"
+      )
+    end
+
+    before do
+      # Extend view with necessary url helpers
+      view.extend BenefitSponsors::Engine.routes.url_helpers
+
+      # Mock EmployerProfile and assign it to view's instance variable
+      employer_profile = instance_double("EmployerProfile", id: 1)
+      assign(:employer_profile, employer_profile)
+
+      # Render the partial with the assigned employer_profile
+      render "ui-components/v1/cards/eligibility_check"
+    end
+
+    it "displays the 'Run Eligibility Check' button" do
+      expect(rendered).to have_selector('input#eligibilityCheckButton[value="Run Eligibility Check"]')
```

Comment:
can we don't use the "value", this implies hardcoding a string, maybe the selector is sufficient